storm-client/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -865,7 +874,7 @@ public static void setupDefaultUncaughtExceptionHandler() {
         Thread.setDefaultUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
                 public void uncaughtException(Thread thread, Throwable thrown) {
                     try {
-                        handleUncaughtException(thrown);
+                        handleUncaughtExceptionWithoutKillingProcess(thrown);
```

Comment:
This will swallow all the `Throwable`s including `Error`s excluding OOME. Is it intended? We may still want to make it crash when there's `Error` thrown from somewhere, given that it normally indicates unrecoverable state.